### PR TITLE
Some minor updates

### PR DESCRIPTION
**Pull request by ivandrofly**
Updated /cc @niksedk

### CODE REVIEW

File: libse/SubtitleFormats/Cavena890.cs [hunk 1]
```diff
@@ -773,9 +775,10 @@ public override void LoadSubtitle(Subtitle subtitle, List<string> lines, string
                 }
                 if (boxType >= 0xa0 && boxType <= 0xa9 && !string.IsNullOrEmpty(p.Text)) // box
                 {
-                    if (p.Text.StartsWith("{\\") && p.Text.Contains("}"))
+                    int endIdx = p.Text.IndexOf('}');
+                    if (p.Text.StartsWith("{\\", StringComparison.Ordinal) && endIdx >= 2)
                     {
-                        p.Text = p.Text.Insert(p.Text.IndexOf("}" + 1, StringComparison.Ordinal), "<box>") + "</box>";
+                        p.Text = p.Text.Insert(endIdx + 1, "<box>") + "</box>";
```
Comment (ivandrofly) [inline libse/SubtitleFormats/Cavena890.cs]: Oops!

File: libse/SubtitleFormats/Cavena890.cs [hunk 2]
```diff
@@ -773,9 +775,10 @@ public override void LoadSubtitle(Subtitle subtitle, List<string> lines, string
                 }
                 if (boxType >= 0xa0 && boxType <= 0xa9 && !string.IsNullOrEmpty(p.Text)) // box
                 {
-                    if (p.Text.StartsWith("{\\") && p.Text.Contains("}"))
+                    int endIdx = p.Text.IndexOf('}');
+                    if (p.Text.StartsWith("{\\", StringComparison.Ordinal) && endIdx >= 2)
                     {
-                        p.Text = p.Text.Insert(p.Text.IndexOf("}" + 1, StringComparison.Ordinal), "<box>") + "</box>";
```
Comment (ivandrofly) [inline libse/SubtitleFormats/Cavena890.cs]: Bug?

File: libse/Utilities.cs [hunk 3]
```diff
@@ -901,14 +901,13 @@ public static bool IsValidRegex(string testPattern)
 
             try
             {
-                Regex.Match(string.Empty, testPattern);
+                return Regex.IsMatch(string.Empty, testPattern);
```
Comment (niksedk) [inline libse/Utilities.cs]: Are you sure this is correct?

Comment (ivandrofly) [inline libse/Utilities.cs]: > Are you sure this is correct?

yeah, Both I'll throw:

**ArgumentException** :  A regular expression parsing error occurred. :)
**ArgumentNullException**: input or pattern is null.
**RegexMatchTimeoutException**: A time-out occurred. For more information about time-outs, see the Remarks section.

* [Regex.Match](https://msdn.microsoft.com/en-us/library/0z2heewz%28v=vs.110%29.aspx)
* [Regex.IsMatch](https://msdn.microsoft.com/en-us/library/sdx2bds0%28v=vs.110%29.aspx)

Also stated in MSDN that:
**The IsMatch method is typically used to validate a string or to ensure that a string conforms to a particular pattern without retrieving that string for subsequent manipulation.**

Comment (niksedk) [inline libse/Utilities.cs]: OK, seems perfect as it is... I've added some unit tests too :)

File: libse/SubtitleFormats/Cavena890.cs [hunk 4]
```diff
@@ -195,15 +195,17 @@ public void Save(string fileName, Subtitle subtitle)
...
                 {
-                    if (p.Text.Contains(new[] { '的', '是', '啊', '吧', '好', '吧', '亲', '爱', '的', '早', '上' }))
+                    if (p.Text.Contains(logoGrams))
                     {
                         _languageIdLine1 = LanguageIdChineseSimplified;
                         _languageIdLine2 = LanguageIdChineseSimplified;
                         break;
                     }
-                    if (p.Text.Contains(new[] { 'я', 'д', 'й', 'л', 'щ', 'ж', 'ц', 'ф', 'ы' }))
+                    if (p.Text.Contains(russianChars))
```
Comment (ivandrofly) [inline libse/SubtitleFormats/Cavena890.cs]: What bout these :arrow_up: ?

Comment (niksedk) [inline libse/SubtitleFormats/Cavena890.cs]: Looks good

Comment (ivandrofly) [inline libse/SubtitleFormats/Cavena890.cs]: > Looks good

:+1:

**Pull request closed without merging**